### PR TITLE
fix #16: Add casting for math database functions

### LINKED ISSUE #16: Unknown signatures for math database functions
There are a number of errors similar to this:
```
======================================================================
ERROR: test_null (db_functions.math.test_sqrt.SqrtTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedFunction: unknown signature: sqrt(int)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/tim/code/django/tests/db_functions/math/test_sqrt.py", line 16, in test_null
    obj = IntegerModel.objects.annotate(null_sqrt=Sqrt('normal')).first()
  File "/home/tim/code/django/django/db/models/query.py", line 653, in first
    for obj in (self if self.ordered else self.order_by('pk'))[:1]:
  File "/home/tim/code/django/django/db/models/query.py", line 274, in __iter__
    self._fetch_all()
  File "/home/tim/code/django/django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/tim/code/django/django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/home/tim/code/django/django/db/models/sql/compiler.py", line 1100, in execute_sql
    cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/tim/code/django/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: unknown signature: sqrt(int)
```
Should I prepare a list of types that cockroachdb's [math and numberic functions](https://www.cockroachlabs.com/docs/stable/functions-and-operators.htm#math-and-numeric-functions) should add support for or should the Django adapter take care of casting based on the currently supported types?

For example, In the latter case, since `tan` supports flat but not decimal or integer, the code would look  like this:
```python
from django.db.models import DecimalField, FloatField, IntegerField
from django.db.models.functions import Cast, Length, Tan

def tan(self, compiler, connection, **extra_context):
    clone = self.copy()
    clone.set_source_expressions([
        Cast(expression, FloatField()) if isinstance(expression.output_field, (DecimalField, IntegerField))
        else expression for expression in self.get_source_expressions()[::-1]
    ])
    return clone.as_sql(compiler, connection, **extra_context)

def register_expressions():
    Tan.as_cockroachdb = tan
```

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

File: cockroach/django/functions.py [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+)
+from django.db.models.functions import (
+    ACos, ASin, ATan, ATan2, Cast, Ceil, Coalesce, Cos, Cot, Degrees, Exp,
+    Floor, Ln, Log, Radians, Round, Sin, Sqrt, StrIndex, Tan,
+)
+
+
+def atan2(self, compiler, connection, **extra_context):
+    # This function is usually ATan2(y, x), returning the inverse tangent
+    # of y / x, but it's ATan2(x, y) on cockroachdb (hence reversed()).
```
Comment (jordanlewis) [inline cockroach/django/functions.py]: Yikes, that is really awful... we probably want to fix that on our side. Thanks for noticing.

Comment (jordanlewis) [reply cockroach/django/functions.py]: Err, I actually think that it's kind of just a documentation fail on our side unless you actually noticed a difference via the tests. The definition was:

atan2(x,y) = `math.Atan2(x, y)`. But `math.Atan2` is defined normally, as `math.Atan2(y,x)` = the inverse tangent of y/x. So I think our argument order was misleadingly reversed, but the result was the same as you'd expect..

Comment (timgraham) [reply cockroach/django/functions.py]: Yes, the tests fail without this fix.

Comment (timgraham) [reply cockroach/django/functions.py]: Just to follow up, there's no issue in cockroachdb. The issue is my bug in the adapter: https://github.com/cockroachdb/cockroach-django/pull/76.

**Pull request merged**